### PR TITLE
[feature] add segmentation modules

### PR DESCRIPTION
**Pull request by Cathy0908**

### CODE REVIEW

File: benchmarks/selfsup/segmentation/voc/fcn_r50-d8_512x512_8xb4_60e_voc12aug.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+# model settings
+norm_cfg = dict(type='SyncBN', requires_grad=True)
+
+model = dict(
+    type='EncoderDecoder',
+    backbone=dict(
+        type='ResNet',
+        depth=50,
+        num_stages=4,
+        # out_indices=(0, 1, 2, 3),
```
Comment (wenmengzhou) [inline benchmarks/selfsup/segmentation/voc/fcn_r50-d8_512x512_8xb4_60e_voc12aug.py:12]: remove this comment

File: benchmarks/selfsup/segmentation/voc/fcn_r50-d8_512x512_8xb4_60e_voc12aug.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                type='SegmentationEvaluator',
+                classes=CLASSES,
+                metric_names=['mIoU'],
+            )
+        ],
+    )
+]
+log_config = dict(
+    interval=50,
+    hooks=[dict(type='TextLoggerHook'),
```
Comment (wenmengzhou) [inline benchmarks/selfsup/segmentation/voc/fcn_r50-d8_512x512_8xb4_60e_voc12aug.py:171]: NIT: segmentation visualization should be supported in the future

File: benchmarks/selfsup/segmentation/voc/fcn_r50-d8_512x512_8xb4_60e_voc12aug.py (hunk 1 above)
Comment (Cathy0908) [reply benchmarks/selfsup/segmentation/voc/fcn_r50-d8_512x512_8xb4_60e_voc12aug.py]: done!

File: benchmarks/selfsup/segmentation/voc/fcn_r50-d8_512x512_8xb4_60e_voc12aug.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    dict(
+        type='Collect',
+        keys=['img', 'gt_semantic_seg'],
+        meta_keys=('filename', 'ori_filename', 'ori_shape', 'img_shape',
+                   'pad_shape', 'scale_factor', 'flip', 'flip_direction',
+                   'img_norm_cfg')),
+]
+test_pipeline = [
+    dict(
+        type='MMMultiScaleFlipAug',
```
Comment (haiasd) [inline benchmarks/selfsup/segmentation/voc/fcn_r50-d8_512x512_8xb4_60e_voc12aug.py:78]: why do data aug in val

**Review by haiasd — APPROVED** [review]

File: easycv/models/loss/cross_entropy_loss.py [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+
+def cross_entropy(pred,
+                  label,
+                  weight=None,
+                  class_weight=None,
+                  reduction='mean',
+                  avg_factor=None,
+                  ignore_index=-100,
+                  avg_non_ignore=False):
+    """cross_entropy. The wrapper function for :func:`F.cross_entropy`
```
Comment (wenmengzhou) [inline easycv/models/loss/cross_entropy_loss.py:94]: describe what is the difference

File: easycv/models/utils/accuracy.py [hunk 5]
```diff
@@ -1,12 +1,26 @@
 # Copyright (c) Alibaba, Inc. and its affiliates.
-import torch.nn as nn
+import torch
 
 
-def accuracy(pred, target, topk=1):
-    """
+def accuracy(pred, target, topk=1, thresh=None, ignore_index=None):
```
Comment (wenmengzhou) [inline easycv/models/utils/accuracy.py:5]: move to core/evaluation/metrics.py ?

File: easycv/datasets/detection/pipelines/mm_transforms.py [hunk 6]
```diff
@@ -1274,8 +1274,9 @@ def __call__(self, results):
 
             # flip segs
             for key in results.get('seg_fields', []):
+                # use copy() to make numpy stride positive
                 results[key] = mmcv.imflip(
-                    results[key], direction=results['flip_direction'])
+                    results[key], direction=results['flip_direction']).copy()
```
[the remark this answers is not in the document]
Comment (Cathy0908) [reply easycv/datasets/detection/pipelines/mm_transforms.py:1279]: not support negetive yet

File: easycv/models/loss/cross_entropy_loss.py (hunk 4 above)
Comment (Cathy0908) [reply easycv/models/loss/cross_entropy_loss.py:94]: done!

**Review by wenmengzhou — APPROVED** [review]
LGTM
